public/app/features/dashboard/components/PanelEditor/state/actions.ts

```diff
@@ -104,9 +104,10 @@ function updateDuplicateLibraryPanels(modifiedPanel: PanelModel, dashboard: Dash
 
     // Loaded plugin is not included in the persisted properties
     // So is not handled by restoreModel
-    panel.plugin = modifiedSaveModel.plugin;
+    const pluginChanged = panel.plugin?.meta.id !== modifiedPanel.plugin?.meta.id;
+    panel.plugin = modifiedPanel.plugin;
 
-    if (panel.type !== modifiedPanel.type) {
+    if (panel.type !== modifiedPanel.type || pluginChanged) {
```

Comment:
not sure I understand this, is this not looking at the same thing? (comparing type) ?

Reply:
By the time this check is executed, `panel.type` should already be equal to `modifiedPanel.type` since we use `restoreModel` and `type` is a persisted property. So I guess the `panel.type !== modifiedPanel.type` bit could actually be removed.